src/compiler/binder.ts

```diff
@@ -2477,6 +2477,11 @@ namespace ts {
 
         function bindSpecialPropertyAssignment(node: BinaryExpression) {
             const lhs = node.left as PropertyAccessEntityNameExpression;
+            // Class declarations in Typescript do not allow property declarations
+            const parentSymbol = lookupSymbolForPropertyAccess(lhs.expression);
+            if (!isInJavaScriptFile(node) && !isTSFunctionSymbol(parentSymbol)) {

```

Comment:
Nit: In both places `isTsFunctionSymbol` is called, you just checked `isInJavaScriptFile` to be false. So maybe it could just be `isFunctionSymbol`.